### PR TITLE
Fix repository for compatibility with --incompatible_no_support_tools_in_action_inputs

### PR DESCRIPTION
**Pull request by laurentlb**

### CODE REVIEW

File: rules/run_binary.bzl [hunk 1]
```diff
@@ -44,7 +44,8 @@ def _impl(ctx):
     }
     ctx.actions.run(
         outputs = ctx.outputs.outs,
-        inputs = depset(direct = ctx.files.srcs, transitive = [tool_inputs]),
+        inputs = depset(direct = ctx.files.srcs),
```
Comment (thomasvl) [inline rules/run_binary.bzl]: Can `inputs` just be ctx.files.srcs or does it have to be wrapped in a depset?

Comment (laurentlb) [reply rules/run_binary.bzl]: Thanks, the depset was indeed useless

File: rules/private/copy_file_private.bzl [hunk 2]
```diff
@@ -48,7 +48,7 @@ def copy_cmd(ctx, src, dst):
 
 def copy_bash(ctx, src, dst):
     ctx.actions.run_shell(
-        inputs = [src],
+        tools = [src],
```
Comment (laszlocsomor) [inline rules/private/copy_file_private.bzl]: Why this change?
I believe we need target-config here, which I believe `inputs` is but `tools` isn't.

Comment (laurentlb) [reply rules/private/copy_file_private.bzl]: https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/116#24ea5562-55a0-41c4-b439-94a288884804

```
Found tool(s) 'bazel-out/host/bin/tests/run_binary/printargs' in inputs. A tool is an input with executable=True set. All tools should be passed using the 'tools' argument instead of 'inputs' in order to make their runfiles available to the action. This safety check will not be performed once the action is modified to take a 'tools' argument. To temporarily disable this check, set --incompatible_no_support_tools_in_action_inputs=false.
```
